src/operation/watch/test.rs

```diff
@@ -0,0 +1,221 @@
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .full_document(FullDocumentType::UpdateLookup)
+        .build();
+
+    let expected_body = doc! {
+        "aggregate": "test_coll",
+        "cursor": {},
+        "pipeline": [doc!{"$changeStream": {"fullDocument": "UpdateLookup"}}, doc!{ "$match": { "x": 3 }}]
+    };
+
+    build_test(
+        ChangeStreamTarget::Collection(ns),
+        pipeline,
+        Some(options),
+        expected_body,
+    );
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn build_cluster() {
+    let pipeline = vec![doc! { "$match": { "x": 3 }}];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .full_document(FullDocumentType::UpdateLookup)
+        .build();
+
+    let expected_body = doc! {
+        "aggregate": 1,
+        "cursor": {},
+        "pipeline": [doc!{"$changeStream": {"fullDocument": "UpdateLookup", "allChangesForCluster": true}},
+        doc!{ "$match": { "x": 3 }}]
+    };
+
+    build_test(
+        ChangeStreamTarget::Cluster("test_db".to_string()),
+        pipeline,
+        Some(options),
+        expected_body,
+    );
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn build_empty_pipeline() {
+    let pipeline = Vec::new();
+    let expected_body = doc! {
+        "aggregate": 1,
+        "cursor": {},
+        "pipeline": [doc!{"$changeStream": {}}]
+    };
+
+    build_test(
+        ChangeStreamTarget::Database("test_db".to_string()),
+        pipeline,
+        None,
+        expected_body,
+    );
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn handle_success() {
+    let ns = Namespace {
+        db: "test_db".to_string(),
+        coll: "test_coll".to_string(),
+    };
+
+    let address = StreamAddress {
+        hostname: "localhost".to_string(),
+        port: None,
+    };
+
+    let pipeline = vec![doc! { "$match": { "x": 3 }}];
+    let watch = Watch::new(
+        ChangeStreamTarget::Database("test_db".to_string()),
+        pipeline.clone(),
+        None,
+    )
+    .unwrap();
+
+    let first_batch = vec![doc! {"_id": 1}, doc! {"_id": 2}];
+
+    let response = doc! {
+        "aggregate": 1,
+        "cursor": {
+            "id": 123,
+            "ns": format!("{}.{}", ns.db, ns.coll),
+            "firstBatch": bson_util::to_bson_array(&first_batch),
+        },
+        "pipeline": [doc!{"$changeStream": {}}, doc!{ "$match": { "x": 3 }}]
+    };
+
+    let result = watch.handle_response(CommandResponse::with_document_and_address(
+        address.clone(),
+        response.clone(),
+    ));
+    assert!(result.is_ok());
+
+    let change_streams_spec = result.unwrap();
```

Comment:
can we drop the "s" from `change_streams_spec`?

Reply:
fixed